### PR TITLE
feat: debounce results

### LINKED ISSUE #3259: [OTHER] Tag/Skill Search Limit on things under 3 characters
### What would you like to share?

Hey Team! :wave: 

First of all, a super cool project I like it a lot and have all my socials pointing to it :smile: 

Just wondering as this is not really a `bug` nor is it really a `feature request` I am just wondering if there is a specific reason behind having a minimum of `3` characters needed for searching? I only ask because I added `Go` as a skill (or `tag`) and it will not let me search on it, but of course `Golang` does.

Not a major issue, I've always just referred to it as `Go` and not `Golang`, but it made me wonder if there is a specific reason other than maybe performance we set a minimum limit to 3 characters. :smiley: 

**P.S** I could also see an issue here if, for example, you want to add `C#`

Thanks, Team!,

### Additional information

Happy to look into this if we think it's worth looking into.

Thanks!

### PR DESCRIPTION
**Pull request by ManishBisht777**
## Fixes Issue #3259 

fixes #3259 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

## typed very fast
image

## Note to reviewers

The first character search always seems to be a bit laggy it makes the page unresponsive for some time maybe thats because it is filtering out the whole 1250+ profiles with just 1 character not sure?

any suggestions? @Parsifal-M @eddiejaoude 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4476"></a>

### CODE REVIEW

File: pages/search.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
-import { useEffect, useState, useRef } from "react";
+import { useEffect, useState, useRef, useCallback } from "react";
 import { useRouter } from "next/router";
 
 import UserCard from "../components/user/UserCard";
 import Alert from "../components/Alert";
 import Page from "../components/Page";
 import PageHead from "../components/PageHead";
 import Tag from "../components/Tag";
+import { debounce } from "lodash";
```
Comment (eddiejaoude) [inline pages/search.js:9]: Is it possible to do it without `lodash`? Maybe with a `setTimeout`

Comment (ManishBisht777) [reply pages/search.js]: Thanks for the suggestion eddie it work better with `setTimeout` learned one more thing today :)
also I attacked a screen recording for the same to

Comment (eddiejaoude) [reply pages/search.js]: Great 👍

**Review by Parsifal-M — APPROVED** [review]
Looks good!

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍

**Comment by eddiejaoude** [comment]
> The first character search always seems to be a bit laggy it makes the page unresponsive for some time maybe thats because it is filtering out the whole 1250+ profiles with just 1 character not sure?

It seemed fine to me. There are a lot of profiles to go through - we can move this to the backend later on in another pull request

**Pull request merged**